client/fingerprint/network_unix.go

```diff
@@ -152,99 +135,83 @@ func (f *NetworkFingerprint) linkSpeedEthtool(path, device string) int {
 	return mbs
 }
 
-// ipAddress returns the first IPv4 address on the configured default interface
-// Tries Golang native functions and falls back onto ifconfig
-func (f *NetworkFingerprint) ipAddress(device string) string {
-	if ip, err := f.nativeIpAddress(device); err == nil {
-		return ip
-	}
-
-	return f.ifConfig(device)
-}
+// Gets the ipv4 addr for a network interface
+func (f *NetworkFingerprint) ipAddress(intf *net.Interface) (string, error) {
```

Comment:
According to the Go style guide in Effetive Go, getters shouldn't have `get` in front: https://golang.org/doc/effective_go.html#Getters